src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    public class PackageAnalyzer : IPackageAnalyzer
+    {
+        private readonly IPackageRestorer _packageRestorer;
+        private readonly IEnumerable<IDependencyAnalyzer> _packageAnalyzers;
+
+        private readonly ILogger _logger;
+
+        public PackageAnalyzer(IPackageRestorer packageRestorer,
+            IEnumerable<IDependencyAnalyzer> packageAnalyzers,
+            ILogger logger)
+        {
+            _packageRestorer = packageRestorer ?? throw new ArgumentNullException(nameof(packageRestorer));
+            _packageAnalyzers = packageAnalyzers ?? throw new ArgumentNullException(nameof(packageAnalyzers));
+            _logger = logger;
+        }
+
+        public async Task<bool> RunPackageAnalyzersAsync(IUpgradeContext context, IDependencyAnalysisState? analysisState, CancellationToken token)
+        {
+            if (context?.CurrentProject is null)
+            {
+                return false;
+            }
+
+            await _packageRestorer.RestorePackagesAsync(context, context.CurrentProject, token).ConfigureAwait(false);
+            var nugetReferences = await context.CurrentProject.GetNuGetReferencesAsync(token).ConfigureAwait(false);
+
+            analysisState = new DependencyAnalysisState(context.CurrentProject, nugetReferences);
```

Comment:
Why is there a parameter to pass this in if it's just set here? Seems like the parameter should be removed

Reply:
analysis state is needed in the packageupdater step and later in the analysis component as well, so unless the analysis state is returned by this method or passed as parameter, how else can it be accessed in packageupdater step?